app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,214 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import { getSubmitterUsersFromCognito } from '~/lib/utils'
+import { getUser } from '../__auth/user.server'
+
+// models
+export type EndorsementRequest = {
+  endorserEmail: string
+  endorserSub: string
+  requestorEmail: string
+  requestorSub: string
+  created: number
+  status: EndorsementStates
+}
+
+export type EndorsementStates = 'approved' | 'pending' | 'rejected' | 'reported'
+
+export class EndorsementsServer {
+  #sub: string
+
+  private constructor(sub: string) {
+    this.#sub = sub
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    if (!user)
+      throw new Response('not signed in', {
+        statusText: 'not signed in',
```

Comment:
I don't think you have to supply both `status` and `statusText`:

```suggestion
```

Reply:
I think that statusText is normally something like "Forbidden" or "Gateway Error". It's the canonical text representation of the HTTP status code. I don't think it's typically used to report the detail of the error.

Unless I'm mistaken, please return the explanatory text in the body of the response, not in statusText.

Reply:
When debugging I wasn't getting any info out of the errors unless there was something in the status text

I moved it for this one, but since the others are all returning otherwise the same type of error I am leaving the string in the statusText

Reply:
What do you mean by "not getting any info"? Where were you expecting to see this message?